witchcraft/wdebug/api.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2025 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wdebug
+
+import (
+	"context"
+	"io"
+)
+
+type DiagnosticType string
```

Comment:
Can this be anything, or are there any guidelines/schema/requirements for the format of values of this type?

Reply:
> Type names must be specific, reasonably expected to be unique, and versioned to allow for future major changes of the payload structure. DiagnosticTypes must match the regular expression ([a-z0-9]+\.)+v[0-9]+, i.e. be lower-case, dot-delimited, and end with a version suffix. For example, the threaddump.v1 diagnosticType might indicate a value of ThreadDumpV1 from the DiagnosticLogV1 definition.

From https://github.palantir.build/deployability/sls-spec/blob/develop/docs/debug.md#diagnostic-types-and-values

Reply:
updated to include a `DiagnosticType.Validate()` method. This method is now called upon wdebug route registration. There is also now a test validating all DiagnosticType constants defined within internal/wdebug